File: perl/ex-001-nightmare-autopsy.pl
#!/usr/bin/perl
# ex-001-sigil-hell.pl AUTOPSY VERSION
# INTENTIONALLY AWFUL: Perl sigil chaos and regex madness
# This celebrates every "write-only" Perl anti-pattern known to humanity
# AUTOPSY: Same nightmare code with detailed explanations of Perl's dark magic

use strict; # PROBLEM: just kidding, we'll break this immediately
use warnings; # PROBLEM: same here

# PROBLEM: Global variable soup with confusing sigils
our ($a,$b,$c,$d,$e,$f,$g,$h,$i,$j,$k,$l,$m,$n,$o,$p,$q,$r,$s,$t,$u,$v,$w,$x,$y,$z); # PROBLEM: 26 global scalars
our @a = qw/a b c d e f g h i j k l m n o p q r s t u v w x y z/; # PROBLEM: Global array with same name
our %a = map { $_ => $_ } @a; # PROBLEM: Global hash with same name - sigil chaos!
our $l1 = 'l1'; our $O0 = 'O0'; our $I1 = 'I1';  # PROBLEM: Confusable variable names
# FIX: Use lexical variables (my), avoid globals, use descriptive names

# PROBLEM: Redefine built-ins for maximum chaos
sub print { CORE::print reverse @_ }  # PROBLEM: Override built-in print function
*CORE::GLOBAL::chomp = sub { $_[0] =~ s/.$//; };  # PROBLEM: Override global chomp to remove last char, not newline
# FIX: Never override built-in functions; use different names

# PROBLEM: The infamous one-liner disasters
$_ = "Hello World"; tr/a-z/A-Z/; s/(.)/reverse $1/ge; print; # PROBLEM: Chained operations on $_, unreadable
# FIX: Use explicit variables, break into multiple lines, add comments

# PROBLEM: Regex hell with maximum backtracking
sub regex_hell {
    my $evil = qr/^(a+)+$/;  # PROBLEM: Catastrophic backtracking - nested quantifiers
    my $input = 'a' x 20 . 'b'; # PROBLEM: Input designed to trigger worst-case performance
    # $input =~ /$evil/;  # PROBLEM: Would hang system for minutes - ReDoS attack!
    
    # PROBLEM: Unreadable regex soup
    my $data = "user:pass@host:port/path?query=value#fragment";
    my ($u,$p,$h,$pt,$pa,$q,$f) = $data =~ m{^([^:]+):([^@]+)@([^:]+):(\d+)/([^?]+)\?([^#]+)#(.+)$}; # PROBLEM: Complex regex in one line
    # FIX: Use named captures, break into parts, validate input
    
    # PROBLEM: Regex as code (eval in disguise)
    my $code = 's/foo/bar/g'; # PROBLEM: String containing regex code
    eval "\$data =~ $code";  # PROBLEM: Dynamic regex execution - code injection risk
    # FIX: Use qr// for precompiled regex, avoid eval with user input
}

# PROBLEM: Sigil switching madness
sub sigil_chaos {
    my @arr = (1,2,3); # PROBLEM: Array variable
    my $arr = [4,5,6]; # PROBLEM: Scalar with same name (array reference)
    my %arr = (7,8,9,10); # PROBLEM: Hash with same name - maximum confusion
    
    print $$arr[0];      # PROBLEM: $arr->[0] but confusing syntax
    print $arr[0];       # PROBLEM: @arr[0] - different variable entirely
    print $arr{7};       # PROBLEM: %arr{7} - yet another different variable
    # FIX: Use descriptive names, avoid sigil reuse, use -> for references
    
    # PROBLEM: Context switching hell
    my $scalar = @arr;   # PROBLEM: Array in scalar context = length (not obvious)
    my @array = %arr;    # PROBLEM: Hash in array context = flattened key-value pairs
    my $ref = \@arr;     # PROBLEM: Reference to array
    my @deref = @$ref;   # PROBLEM: Dereference array reference
    # FIX: Be explicit about context, use scalar(@arr) for clarity
}

# PROBLEM: Special variables abuse
sub special_vars_hell {
    $" = '|';           # PROBLEM: Array separator - affects all array interpolation
    $, = '::';          # PROBLEM: Output separator for print
    $\ = "\n---\n";     # PROBLEM: Output record separator - added to all prints
    $/ = undef;         # PROBLEM: Slurp mode - reads entire files at once
    $. = 999;           # PROBLEM: Line number counter manipulation
    $? = 42;            # PROBLEM: Child process exit status manipulation
    # FIX: Use explicit formatting, avoid global special variables
    
    # PROBLEM: The infamous $_ abuse
    for (1..10) {       # PROBLEM: $_ is implicit loop variable
        $_ *= 2;        # PROBLEM: Modify $_ directly
        s/(\d+)/$1 bottles/; # PROBLEM: Regex operates on $_ implicitly
        print;          # PROBLEM: print operates on $_ implicitly
    }
    # FIX: Use explicit loop variables: for my $i (1..10)
    
    # PROBLEM: Punctuation variables - cryptic and unreadable
    print $0, $!, $@, $$, $<, $>, $(, $), $[, $], $^T, $^O; # PROBLEM: Program name, error, eval error, PID, UID, etc.
    # FIX: Use English module: use English; then $PROGRAM_NAME instead of $0
}

# PROBLEM: Reference and typeglob chaos
sub reference_hell {
    my $scalar = "test";
    my @array = (1,2,3);
    my %hash = (a => 1, b => 2);
    
    # PROBLEM: Typeglob manipulation - advanced Perl dark magic
    *foo = \$scalar;     # PROBLEM: Assign scalar reference to typeglob
    *foo = \@array;      # PROBLEM: Now foo is array - same symbol table entry
    *foo = \%hash;       # PROBLEM: Now foo is hash - completely different meaning
    # FIX: Use explicit references, avoid typeglobs unless absolutely necessary
    
    # PROBLEM: Anonymous reference madness
    my $complex = {      # PROBLEM: Deeply nested anonymous structures
        arr => [1, 2, { nested => [qw/a b c/] }], # PROBLEM: Array of mixed types
        code => sub { $_[0] * 2 }, # PROBLEM: Anonymous subroutine reference
        regex => qr/\d+/   # PROBLEM: Compiled regex reference
    };
    
    # PROBLEM: Dereference hell - hard to read and understand
    print @{$complex->{arr}}; # PROBLEM: Dereference array reference
    print ${$complex->{arr}}[2]->{nested}->[1]; # PROBLEM: Deep dereferencing chain
    print $complex->{code}->(42); # PROBLEM: Call code reference
    # FIX: Use intermediate variables, avoid deep nesting
}

# PROBLEM: Package and namespace pollution
package Chaos; # PROBLEM: Switch to different package
our @ISA = qw/UNIVERSAL/;  # PROBLEM: Inherit from UNIVERSAL (everything)
sub AUTOLOAD {  # PROBLEM: AUTOLOAD catches all undefined method calls
    our $AUTOLOAD; # PROBLEM: Global variable with method name
    print "Called: $AUTOLOAD\n";
    goto &{$AUTOLOAD};  # PROBLEM: Infinite recursion potential - will crash
}
# FIX: Use proper OOP with Moose/Moo, avoid AUTOLOAD, explicit inheritance

package main; # PROBLEM: Switch back to main package

# PROBLEM: Prototype abuse
sub mysub ($@%) { }  # PROBLEM: Confusing prototypes that change calling conventions
sub myprint ($) { print $_[0] } # PROBLEM: Prototype forces scalar context
# FIX: Avoid prototypes unless you understand them completely

# PROBLEM: File handle chaos
sub file_hell {
    open F, ">", "/tmp/chaos.txt" or die;  # PROBLEM: Global filehandle, no lexical scoping
    print F "chaos\n"; # PROBLEM: Print to global filehandle
    # PROBLEM: No close() - file handle leak
    
    # PROBLEM: Bareword file handles
    open STDOUT, ">&STDERR";  # PROBLEM: Redirect stdout to stderr globally
    
    # PROBLEM: Slurp without error checking
    my $content = do { local $/; <F> }; # PROBLEM: Slurp mode, no error handling
    # FIX: Use lexical filehandles: open my $fh, '>', $file; always close; check errors
}

# PROBLEM: Eval and string execution
sub eval_hell {
    my $code = 'print "Hello from eval\n"'; # PROBLEM: Code as string
    eval $code;  # PROBLEM: String eval - code injection vulnerability
    # FIX: Use block eval: eval { ... } for exception handling only
    
    # PROBLEM: Symbolic references (soft refs)
    my $var = "test"; # PROBLEM: Variable name as string
    $$var = "chaos";  # PROBLEM: Creates $test = "chaos" - symbolic reference
    # FIX: Use hash references or proper data structures
    
    # PROBLEM: Dynamic subroutine calls
    my $sub = "regex_hell"; # PROBLEM: Subroutine name as string
    &$sub();  # PROBLEM: Call subroutine by name - can be exploited
    # FIX: Use code references: my $sub = \&regex_hell; $sub->();
}

# PROBLEM: Format chaos (ancient Perl feature)
format STDOUT = # PROBLEM: Ancient formatting system, hard to maintain
@<<<<<<< @|||||| @>>>>>>
$a,      $b,     $c
.
# FIX: Use printf, sprintf, or template systems

# PROBLEM: Main chaos orchestrator
sub main {
    print "🐪 PERL NIGHTMARE STARTING 🐪\n";
    
    regex_hell();           # PROBLEM: Regex catastrophic backtracking
    sigil_chaos();          # PROBLEM: Sigil confusion and context switching
    special_vars_hell();    # PROBLEM: Global special variable abuse
    reference_hell();       # PROBLEM: Complex reference and typeglob manipulation
    file_hell();            # PROBLEM: File handle leaks and global handles
    eval_hell();            # PROBLEM: String eval and symbolic references
    
    # PROBLEM: The ultimate Perl one-liner disasters
    map{print$_}sort{$a<=>$b}grep{/\d/}split//,join'',<>; # PROBLEM: Completely unreadable
    # TRANSLATION: Read input, join lines, split into chars, grep digits, sort numerically, print each
    
    # PROBLEM: Obfuscated code golf
    $_='987654321';s/./print"$& "x$&,"\n"/ge; # PROBLEM: Print each digit N times
    # TRANSLATION: For each digit, print it that many times
    
    print "🎭 PERL CHAOS COMPLETE 🎭\n";
}

# PROBLEM: Execute with implicit calls
main; # PROBLEM: Call without parentheses

# ─────────────────────────────────────────────────────────────────────────────
# PERL ANTI-PATTERNS SUMMARY:
# ─────────────────────────────────────────────────────────────────────────────
# 1. **Sigil Chaos**: Same name with different sigils ($, @, %) means different variables
# 2. **Special Variables**: Cryptic punctuation variables ($!, $@, $$, etc.)
# 3. **Context Sensitivity**: Same expression means different things in different contexts
# 4. **Global Variables**: Everything is global by default, use 'my' for lexical scope
# 5. **Regex Complexity**: Complex regex in one line, catastrophic backtracking
# 6. **String Eval**: eval with strings allows code injection
# 7. **Symbolic References**: Using strings as variable names
# 8. **Typeglob Manipulation**: Advanced feature that's usually unnecessary
# 9. **One-liners**: Compressed code that's impossible to read/maintain
# 10. **Implicit $_**: Many operations work on $_ implicitly

# WHY PERL IS UNIQUELY CHAOTIC:
# ─────────────────────────────────────────────────────────────────────────────
# - **TIMTOWTDI**: "There's More Than One Way To Do It" - too much flexibility
# - **Context Sensitivity**: Same code behaves differently in scalar/list context
# - **Sigil System**: $, @, % prefixes change meaning of same identifier
# - **Special Variables**: Dozens of cryptic built-in variables
# - **Regex Integration**: Regular expressions are first-class language feature
# - **Global by Default**: Variables are global unless explicitly made lexical
# - **Implicit $_**: Many operations work on default variable $_
# - **Write-Only Code**: Easy to write, impossible to read later

# FIX SUMMARY:
# ─────────────────────────────────────────────────────────────────────────────
# 1. Always use 'strict' and 'warnings' pragmas
# 2. Use 'my' for lexical variables, avoid globals
# 3. Use descriptive variable names, avoid sigil reuse
# 4. Break complex regex into parts with /x modifier
# 5. Use English module for readable special variables
# 6. Avoid string eval, use block eval for exceptions
# 7. Use explicit variables instead of $_
# 8. Comment complex one-liners or break them up
# 9. Use modern Perl features (Moose, Try::Tiny, etc.)
# 10. Follow Perl Best Practices guidelines

# Remember: Perl's motto "There's more than one way to do it" is both 
# its strength and weakness! Choose the readable way! 🐪

File: php/ex-001-nightmare-autopsy.php
<?php
// ex-001-variable-variable-challenge.php AUTOPSY VERSION
// INTENTIONALLY AWFUL: PHP variable variables and magic method madness
// This file celebrates every PHP WTF moment and anti-pattern known to humanity
// AUTOPSY: Same nightmare code with detailed explanations of PHP's dangerous features

// PROBLEM: Error suppression paradise - silence ALL the warnings
error_reporting(0);                                              // PROBLEM: Disable all error reporting
ini_set('display_errors', 0);                                   // PROBLEM: Hide errors from display
// FIX: Enable error reporting in development, use proper error handling

// PROBLEM: Global state soup with confusing variable names
$GLOBALS['chaos'] = 'global chaos';                             // PROBLEM: Global variable pollution
$GLOBALS['l1'] = 'l1';                                          // PROBLEM: Confusable with I1
$GLOBALS['O0'] = 'O0';                                          // PROBLEM: Confusable with 00
$GLOBALS['I1'] = 'I1';                                          // PROBLEM: Confusable with l1
$GLOBALS['data'] = array();                                     // PROBLEM: Generic global array
$GLOBALS['config'] = array();                                  // PROBLEM: Global config array
$GLOBALS['state'] = array();                                   // PROBLEM: Global state array
$GLOBALS['db'] = null;                                         // PROBLEM: Global database connection
$GLOBALS['user'] = null;                                       // PROBLEM: Global user object
// FIX: Use classes, dependency injection, avoid global state

// PROBLEM: Variable variables chaos
$var = 'chaos';                                                // PROBLEM: Variable name as string
$$var = 'variable variable madness';                           // PROBLEM: $chaos = 'variable variable madness'
$$$var = 'triple variable chaos';                              // PROBLEM: ${$chaos} = 'triple variable chaos'
// FIX: Use arrays or objects instead of variable variables

// PROBLEM: Dynamic variable names from user input (NEVER DO THIS)
function createVariableFromInput($input) {
    global ${$input};                                           // PROBLEM: Create global variable from user input
    ${$input} = "HACKED: " . $input;                           // PROBLEM: Assign value to dynamic variable
    
    // PROBLEM: Even worse - create function names from input
    $funcName = 'process_' . $input;                           // PROBLEM: Function name from user input
    if (!function_exists($funcName)) {
        eval("function $funcName() { return 'EVAL CHAOS'; }"); // PROBLEM: EVAL IS EVIL - code injection
    }
    
    return ${$input};
}
// FIX: Validate and sanitize input, use whitelists, avoid eval()

// PROBLEM: Magic method overuse
class ChaosClass {
    private $data = array();                                    // PROBLEM: Private data accessed via magic methods
    private $config = array();                                  // PROBLEM: Private config accessed via magic methods
    
    // PROBLEM: Magic method abuse
    public function __get($name) {
        // PROBLEM: Return anything for any property
        if (isset($this->data[$name])) {
            return $this->data[$name];                          // PROBLEM: Return unvalidated data
        }
        
        // PROBLEM: Create properties on the fly
        $this->data[$name] = "MAGIC_" . strtoupper($name);      // PROBLEM: Auto-create properties
        return $this->data[$name];
    }
    
    public function __set($name, $value) {
        // PROBLEM: Set anything as any property
        $this->data[$name] = $value;                            // PROBLEM: No validation or type checking
        
        // PROBLEM: Side effects in __set
        global $GLOBALS;                                        // PROBLEM: Access globals in magic method
        $GLOBALS['last_set'] = $name;                          // PROBLEM: Global side effect
        $GLOBALS['set_count'] = isset($GLOBALS['set_count']) ? $GLOBALS['set_count'] + 1 : 1;
    }
    
    public function __call($method, $args) {
        // PROBLEM: Handle any method call
        if (strpos($method, 'get') === 0) {                     // PROBLEM: String-based method detection
            $property = strtolower(substr($method, 3));         // PROBLEM: Extract property name
            return $this->__get($property);                     // PROBLEM: Delegate to magic __get
        }
        
        if (strpos($method, 'set') === 0) {                     // PROBLEM: String-based method detection
            $property = strtolower(substr($method, 3));         // PROBLEM: Extract property name
            $this->__set($property, isset($args[0]) ? $args[0] : null);
            return $this;                                       // PROBLEM: Return $this for chaining
        }
        
        // PROBLEM: Execute arbitrary code based on method name
        if (method_exists($this, 'handle_' . $method)) {        // PROBLEM: Dynamic method calling
            return call_user_func_array(array($this, 'handle_' . $method), $args);
        }
        
        // PROBLEM: Last resort - eval the method name
        $code = "return 'DYNAMIC_METHOD_" . strtoupper($method) . "';";
        return @eval($code);                                    // PROBLEM: Eval with error suppression
    }
    
    public function __toString() {
        // PROBLEM: Return serialized data (could expose sensitive info)
        return serialize($this->data);                          // PROBLEM: Serialize private data
    }
    
    public function __invoke() {
        // PROBLEM: Make object callable
        return "OBJECT_INVOKED_" . time();                      // PROBLEM: Time-based return value
    }
    
    public function __debugInfo() {
        // PROBLEM: Expose everything during debugging
        return array_merge($this->data, $this->config, $GLOBALS); // PROBLEM: Expose globals and private data
    }
}
// FIX: Use explicit methods, validate inputs, avoid magic methods when possible

// PROBLEM: SQL injection paradise
function getUserData($username, $password) {
    global $db;
    
    // PROBLEM: Direct string concatenation - SQL injection heaven
    $query = "SELECT * FROM users WHERE username = '" . $username . "' AND password = '" . $password . "'";
    
    // PROBLEM: Log the query with sensitive data
    error_log("Executing query: " . $query);                   // PROBLEM: Log passwords in plain text
    
    // PROBLEM: Execute without any validation
    $result = @mysql_query($query, $db);                       // PROBLEM: Deprecated mysql_ functions
    
    if ($result) {
        $user = mysql_fetch_assoc($result);                     // PROBLEM: Deprecated function
        
        // PROBLEM: Store sensitive data in global state
        $GLOBALS['current_user'] = $user;                       // PROBLEM: User data in globals
        $GLOBALS['last_query'] = $query;                        // PROBLEM: Query with passwords in globals
        
        return $user;
    }
    
    return false;
}
// FIX: Use prepared statements, PDO, never log sensitive data, validate inputs

// PROBLEM: Include/require hell with circular dependencies
function loadConfig($configName) {
    global $config;
    
    // PROBLEM: Dynamic includes based on user input
    $configFile = "config/" . $configName . ".php";            // PROBLEM: Path from user input
    
    // PROBLEM: No validation of file path
    if (@file_exists($configFile)) {                           // PROBLEM: Error suppression
        include $configFile;                                    // PROBLEM: Could include anything
    } else {
        // PROBLEM: Try alternative paths
        include_once "configs/" . $configName . ".php";        // PROBLEM: More dynamic includes
        require "settings/" . $configName . ".config.php";     // PROBLEM: Require without error handling
        require_once "../" . $configName . ".php";             // PROBLEM: Directory traversal possible
    }
    
    // PROBLEM: Circular dependency - config files that include this file
    if (isset($config['dependencies'])) {
        foreach ($config['dependencies'] as $dep) {
            loadConfig($dep);                                   // PROBLEM: Potential infinite recursion
        }
    }
}
// FIX: Validate file paths, use whitelists, avoid circular dependencies

// PROBLEM: Error suppression abuse
function errorSuppressionParadise() {
    // PROBLEM: Suppress ALL the errors
    @file_get_contents("http://nonexistent-domain.com/api");    // PROBLEM: Suppress network errors
    @json_decode("invalid json");                               // PROBLEM: Suppress JSON errors
    @mysql_connect("invalid_host", "user", "pass");            // PROBLEM: Suppress connection errors
    @fopen("/root/secret.txt", "r");                           // PROBLEM: Suppress file access errors
    @unlink("/important/system/file");                         // PROBLEM: Suppress file deletion errors
    
    // PROBLEM: Suppress division by zero
    $result = @(10 / 0);                                       // PROBLEM: Suppress math errors
    
    // PROBLEM: Suppress array access errors
    $data = array();
    $value = @$data['nonexistent']['nested']['key'];           // PROBLEM: Suppress undefined index
    
    // PROBLEM: Suppress function call errors
    @nonExistentFunction("with", "parameters");                // PROBLEM: Suppress undefined function
    
    return "All errors suppressed successfully!";
}
// FIX: Handle errors properly, use try-catch, validate before operations

// PROBLEM: Global variable mutation everywhere
function mutateGlobals() {
    global $chaos, $data, $config, $state, $user;              // PROBLEM: Import many globals
    
    // PROBLEM: Modify globals without any coordination
    $chaos .= "_MUTATED";                                      // PROBLEM: String concatenation to global
    $data['mutated'] = true;                                   // PROBLEM: Add to global array
    $config = array_merge($config, array('chaos' => true));   // PROBLEM: Merge into global config
    $state['mutations'] = isset($state['mutations']) ? $state['mutations'] + 1 : 1;
    
    // PROBLEM: Create new globals dynamically
    $GLOBALS['dynamic_global_' . time()] = "CREATED_AT_RUNTIME"; // PROBLEM: Runtime global creation
    
    // PROBLEM: Modify superglobals
    $_SESSION['chaos'] = true;                                  // PROBLEM: Modify session
    $_COOKIE['hacked'] = 'yes';                                // PROBLEM: Modify cookies (won't work)
    $_SERVER['CUSTOM_CHAOS'] = 'injected';                    // PROBLEM: Modify server vars
}
// FIX: Use classes, dependency injection, avoid global mutation

// PROBLEM: Type juggling chaos
function typeJugglingChaos($input) {
    // PROBLEM: PHP's loose comparison madness
    if ($input == 0) {                                         // PROBLEM: Loose comparison
        echo "Input equals 0\n";                               // PROBLEM: True for "", "0", false, null, array()
    }
    
    if ($input == false) {                                     // PROBLEM: Loose comparison
        echo "Input equals false\n";                           // PROBLEM: True for 0, "", "0", null, array()
    }
    
    if ($input == null) {                                      // PROBLEM: Loose comparison
        echo "Input equals null\n";                            // PROBLEM: True for false, 0, "", "0"
    }
    
    // PROBLEM: String to number conversion chaos
    $result = $input + 0;                                      // PROBLEM: Force numeric conversion
    $stringResult = $input . "";                               // PROBLEM: Force string conversion
    
    // PROBLEM: Array to string conversion
    if (is_array($input)) {
        $arrayString = (string)$input;                          // PROBLEM: Just returns "Array"
    }
    
    return array(
        'original' => $input,
        'as_number' => $result,
        'as_string' => $stringResult,
        'type' => gettype($input)
    );
}
// FIX: Use strict comparisons (===), validate types explicitly

// PROBLEM: File handling disasters
function fileHandlingChaos($filename) {
    // PROBLEM: Open file without checking if it exists or is readable
    $handle = @fopen($filename, "r+");                         // PROBLEM: Suppress errors, read+write mode
    
    if ($handle) {
        // PROBLEM: Read entire file into memory without size limits
        $content = fread($handle, filesize($filename) * 2);    // PROBLEM: Read more than file size
        
        // PROBLEM: Write without checking permissions
        fwrite($handle, "CHAOS_INJECTED_" . time());           // PROBLEM: Inject content into file
        
        // PROBLEM: Don't close the handle - resource leak
        // fclose($handle);                                    // PROBLEM: Commented out for chaos
    }
    
    // PROBLEM: File operations without error checking
    @copy($filename, "/tmp/backup_" . basename($filename));    // PROBLEM: Copy without validation
    @chmod($filename, 0777);                                   // PROBLEM: World writable permissions
    @chown($filename, "www-data");                             // PROBLEM: Change ownership
    
    // PROBLEM: Include user-provided files
    if (strpos($filename, '.php') !== false) {                 // PROBLEM: Weak file type check
        @include $filename;                                     // PROBLEM: Code injection vulnerability
    }
    
    return "File chaos complete";
}
// FIX: Validate file paths, check permissions, use proper error handling

// PROBLEM: Session chaos
function sessionChaos() {
    // PROBLEM: Start session without proper configuration
    @session_start();                                           // PROBLEM: Suppress session errors
    
    // PROBLEM: Store sensitive data in session
    $_SESSION['password'] = 'admin123';                        // PROBLEM: Password in session
    $_SESSION['credit_card'] = '4111-1111-1111-1111';         // PROBLEM: Credit card in session
    $_SESSION['ssn'] = '123-45-6789';                          // PROBLEM: SSN in session
    
    // PROBLEM: Session fixation vulnerability
    if (isset($_GET['session_id'])) {                          // PROBLEM: Accept session ID from GET
        session_id($_GET['session_id']);                       // PROBLEM: Set session ID from user input
        session_start();                                        // PROBLEM: Start with user-provided ID
    }
    
    // PROBLEM: Store objects in session without proper serialization
    $_SESSION['chaos_object'] = new ChaosClass();              // PROBLEM: Object in session
    
    // PROBLEM: Never regenerate session ID
    // session_regenerate_id();                                // PROBLEM: Commented out for chaos
}
// FIX: Configure sessions properly, don't store sensitive data, regenerate IDs

// PROBLEM: Output buffering chaos
function outputBufferingChaos() {
    // PROBLEM: Nested output buffers
    ob_start();                                                 // PROBLEM: Start buffer level 1
    echo "Level 1\n";
    
    ob_start();                                                 // PROBLEM: Start buffer level 2
    echo "Level 2\n";
    
    ob_start();                                                 // PROBLEM: Start buffer level 3
    echo "Level 3\n";
    
    // PROBLEM: Mix ob_get_contents and ob_end_clean randomly
    $level3 = ob_get_contents();                               // PROBLEM: Get level 3 contents
    ob_end_clean();                                            // PROBLEM: Clean level 3
    
    echo "Back to level 2\n";
    $level2 = ob_get_contents();                               // PROBLEM: Get level 2 contents
    // PROBLEM: Don't clean level 2 - leave it hanging
    
    echo "Back to level 1\n";
    $level1 = ob_get_contents();                               // PROBLEM: Get level 1 contents
    ob_end_flush();                                            // PROBLEM: Flush instead of clean
    
    return array($level1, $level2, $level3);
}
// FIX: Manage output buffers properly, clean up all levels

// PROBLEM: Main chaos orchestrator
function main() {
    echo "🐘 PHP NIGHTMARE STARTING - PREPARE FOR CHAOS! 🐘\n";
    
    // PROBLEM: Initialize global chaos
    global $chaos, $data, $config, $state;                     // PROBLEM: Import globals
    $data = array('initialized' => true);                      // PROBLEM: Initialize global
    $config = array('chaos_mode' => true);                     // PROBLEM: Initialize global config
    $state = array('active' => true);                          // PROBLEM: Initialize global state
    
    // PROBLEM: Variable variable chaos
    $result1 = createVariableFromInput("user_input");          // PROBLEM: Create variable from string
    echo "Variable variable result: $result1\n";
    
    // PROBLEM: Magic method chaos
    $chaosObj = new ChaosClass();                              // PROBLEM: Create chaos object
    $chaosObj->dynamicProperty = "DYNAMIC_VALUE";              // PROBLEM: Use magic __set
    echo "Magic property: " . $chaosObj->dynamicProperty . "\n"; // PROBLEM: Use magic __get
    echo "Magic method: " . $chaosObj->getDynamicData("param") . "\n"; // PROBLEM: Use magic __call
    echo "Object as string: " . $chaosObj . "\n";             // PROBLEM: Use magic __toString
    echo "Invoke object: " . $chaosObj() . "\n";              // PROBLEM: Use magic __invoke
    
    // PROBLEM: SQL injection chaos
    $userData = getUserData("admin' OR '1'='1", "password");   // PROBLEM: SQL injection payload
    if ($userData) {
        echo "User data retrieved (probably hacked)\n";
    }
    
    // PROBLEM: Include chaos
    loadConfig("database");                                     // PROBLEM: Might cause circular includes
    
    // PROBLEM: Error suppression chaos
    $suppressionResult = errorSuppressionParadise();           // PROBLEM: Suppress all errors
    echo "Error suppression: $suppressionResult\n";
    
    // PROBLEM: Global mutation chaos
    mutateGlobals();                                           // PROBLEM: Mutate global state
    echo "Globals mutated\n";
    
    // PROBLEM: Type juggling chaos
    $typeResults = array(                                      // PROBLEM: Array of type juggling results
        typeJugglingChaos("0"),                               // PROBLEM: String "0"
        typeJugglingChaos(false),                             // PROBLEM: Boolean false
        typeJugglingChaos(null),                              // PROBLEM: Null value
        typeJugglingChaos(array()),                           // PROBLEM: Empty array
        typeJugglingChaos("")                                 // PROBLEM: Empty string
    );
    echo "Type juggling results: " . print_r($typeResults, true) . "\n";
    
    // PROBLEM: File handling chaos
    $fileResult = fileHandlingChaos("/tmp/chaos.txt");         // PROBLEM: File operations without validation
    echo "File chaos: $fileResult\n";
    
    // PROBLEM: Session chaos
    sessionChaos();                                            // PROBLEM: Session mismanagement
    echo "Session chaos complete\n";
    
    // PROBLEM: Output buffering chaos
    $bufferResults = outputBufferingChaos();                   // PROBLEM: Nested buffer chaos
    echo "Buffer chaos: " . print_r($bufferResults, true) . "\n";
    
    // PROBLEM: Final chaos report
    echo "🔥 PHP CHAOS COMPLETE! 🔥\n";
    echo "Global variables: " . count($GLOBALS) . "\n";       // PROBLEM: Count globals
    echo "Memory usage: " . memory_get_usage() . " bytes\n";  // PROBLEM: Memory usage
    echo "Peak memory: " . memory_get_peak_usage() . " bytes\n"; // PROBLEM: Peak memory
    
    // PROBLEM: Exit without cleanup
    exit(0);                                                   // PROBLEM: Force exit
}

// PROBLEM: Auto-execute chaos
main();                                                        // PROBLEM: Execute immediately

// PROBLEM: Dead code that might still execute due to includes
if (false) {                                                   // PROBLEM: Dead code block
    echo "This should never execute\n";
    eval($_POST['code']);                                      // PROBLEM: Remote code execution
    system($_GET['cmd']);                                      // PROBLEM: Command injection
}

?>

<!-- PROBLEM: HTML mixed with PHP for extra chaos -->
<script>
// PROBLEM: JavaScript in PHP file
var phpData = <?php echo json_encode($GLOBALS); ?>;           // PROBLEM: Expose PHP globals to JS
console.log('PHP globals exposed to JavaScript:', phpData);

// PROBLEM: XSS vulnerability
document.write('<?php echo $_GET["xss"]; ?>');                // PROBLEM: Direct output of GET parameter
</script>

<?php
// PROBLEM: More PHP after HTML
echo "<!-- This comment contains sensitive data: " . serialize($_SESSION) . " -->\n"; // PROBLEM: Expose session in HTML comment
?>

<?php
// ─────────────────────────────────────────────────────────────────────────────
// PHP ANTI-PATTERNS SUMMARY:
// ─────────────────────────────────────────────────────────────────────────────
// 1. **Variable Variables**: Using $$var creates dynamic variable names
// 2. **Magic Method Overuse**: __get, __set, __call bypass normal method calls
// 3. **SQL Injection**: String concatenation in SQL queries
// 4. **Error Suppression**: @ operator hides all errors and warnings
// 5. **Global State Abuse**: Excessive use of global variables and $GLOBALS
// 6. **Include/Require Hell**: Dynamic includes with user input
// 7. **Type Juggling**: Loose comparisons (==) cause unexpected behavior
// 8. **Eval() Usage**: eval() executes arbitrary code - major security risk
// 9. **Session Mismanagement**: Storing sensitive data, session fixation
// 10. **XSS Vulnerabilities**: Direct output of user input without escaping

// WHY PHP IS UNIQUELY DANGEROUS:
// ─────────────────────────────────────────────────────────────────────────────
// - **Variable Variables**: Can create variables from user input
// - **Magic Methods**: Can intercept and modify all property/method access
// - **Type Juggling**: Automatic type conversion causes unexpected comparisons
// - **Global State**: Easy access to global variables and superglobals
// - **Include System**: Can include any file, even user-provided ones
// - **Error Suppression**: @ operator can hide critical errors
// - **Mixed HTML/PHP**: Easy to create XSS vulnerabilities
// - **Backwards Compatibility**: Many deprecated and dangerous functions still work

// FIX SUMMARY:
// ─────────────────────────────────────────────────────────────────────────────
// 1. Avoid variable variables, use arrays or objects instead
// 2. Use explicit methods instead of magic methods when possible
// 3. Use prepared statements (PDO) for database queries
// 4. Handle errors properly, don't suppress with @
// 5. Use classes and dependency injection instead of globals
// 6. Validate and whitelist file includes, avoid dynamic includes
// 7. Use strict comparisons (===) instead of loose (==)
// 8. Never use eval(), validate and sanitize all input
// 9. Configure sessions properly, don't store sensitive data
// 10. Escape output with htmlspecialchars(), validate all input

// Remember: PHP gives you many ways to shoot yourself in the foot - be careful! 🐘
?>
